scipy/linalg/tests/test_lapack.py

```diff
@@ -1705,3 +1705,183 @@ def test_getc2_gesc2():
         else:
             assert_array_almost_equal(desired_cplx.astype(dtype),
                                       x/scale, decimal=4)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize("dtype", DTYPES)
+@pytest.mark.parametrize("fact,dl_d_du_lambda",
+                         [("F", lambda dl, d, du: get_lapack_funcs('gttrf',
+                                                                   dtype=d.dtype)),
+                          ("N", lambda dl, d, du: (None, None, None, None,
+                                                   None))])
+def test_gtsvx(dtype, fact, dl_d_du_lambda):
+    """
+    This test uses ?gtsvx to solve a random Ax=b system for each dtype.
+    It tests that the outputs define an LU matrix, that inputs are unmodified,
+    transposal options, incompatible shapes, singular matricies, and
+    singular factorizations. It parametrizes DTYPES and the fact value along
+    with the fact related inputs.
+    """
+    np.random.seed(42)
+    # set test tolerance appropriate for dtype
+    rtol = 250 * np.finfo(dtype).eps
+    atol = 100 * np.finfo(dtype).eps
+    # obtain routine
+    gtsvx = get_lapack_funcs('gtsvx', dtype=dtype)
+    # n is the length diagonal of A
+    n = 10
+    # generate facots dl, d, and du.
+    dl = generate_random_dtype_array((n-1,), dtype=dtype)
+    d = generate_random_dtype_array((n,), dtype=dtype)
+    du = generate_random_dtype_array((n-1,), dtype=dtype)
+    # create A from these diagonals
+    A = np.diag(dl, -1) + np.diag(d) + np.diag(du, 1)
+    # generate random solution x
+    x = generate_random_dtype_array((n, 2), dtype=dtype)
+    # create b from x for equation Ax=b
+    b = A @ x
+    inputs_cpy = [dl.copy(), d.copy(), du.copy(), b.copy()]
```

Comment:
On the other hand, for a line like _this_ a comment would be helpful since you don't use `inputs_cpy` immediately.